awx/main/tasks/jobs.py

```diff
@@ -76,6 +76,177 @@
 logger = logging.getLogger('awx.main.tasks.jobs')
 
 
+class RunnerCallback:
```

Comment:
I don't see this used anywhere?

You have references like `self.event_data_key` in here. In that, `self` is supposed to be a subclass of `BaseTask` and that's not true when you split it into a different class. So you can process these in an `__init__` method, and if it makes sense to you, save it right to `self` again. This clarifies the data flow between the task class and the callback class.

And obviously, you'll have to initialize it somewhere and replace the references to the methods.

Reply:
@AlanCoding , thank you.
I have tried to make the required changes.